test/e2e/specs/editor/various/embedding.spec.js

```diff
@@ -0,0 +1,310 @@
+	code: 'oembed_invalid_url',
+	message: 'Not Found',
+	data: {
+		status: 404,
+	},
+	html: false,
+};
+
+test.use( {
+	embedUtils: async ( { page, editor }, use ) => {
+		await use( new EmbedUtils( { page, editor } ) );
+	},
+} );
+
+test.describe( 'Embedding content', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should render embeds in the correct state', async ( {
+		editor,
+		embedUtils,
+	} ) => {
+		await embedUtils.interceptRequests( {
+			'https://twitter.com/notnownikki': MOCK_EMBED_RICH_SUCCESS_RESPONSE,
+			'https://twitter.com/wooyaygutenberg123454312':
+				MOCK_CANT_EMBED_RESPONSE,
+			'https://wordpress.org/gutenberg/handbook/':
+				MOCK_BAD_WORDPRESS_RESPONSE,
+			'https://twitter.com/thatbunty': MOCK_BAD_EMBED_PROVIDER_RESPONSE,
+			'https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/':
+				MOCK_EMBED_WORDPRESS_SUCCESS_RESPONSE,
+			'https://www.youtube.com/watch?v=lXMskKTw3Bc':
+				MOCK_EMBED_VIDEO_SUCCESS_RESPONSE,
+			'https://cloudup.com/cQFlxqtY4ob':
+				MOCK_EMBED_PHOTO_SUCCESS_RESPONSE,
+		} );
+
+		const currenEmbedBlock = editor.canvas
+			.getByRole( 'document', { name: 'Block' } )
+			.last();
+
+		await embedUtils.insertEmbed( 'https://twitter.com/notnownikki' );
+		await expect(
+			currenEmbedBlock.locator( 'iframe' ),
+			'Valid embed. Should render valid element.'
+		).toHaveAttribute( 'title', 'Embedded content from twitter' );
+
+		await embedUtils.insertEmbed(
+			'https://twitter.com/wooyaygutenberg123454312'
+		);
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'Valid provider; invalid content. Should render failed, edit state.'
+		).toHaveValue( 'https://twitter.com/wooyaygutenberg123454312' );
+
+		await embedUtils.insertEmbed(
+			'https://wordpress.org/gutenberg/handbook/'
+		);
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'WordPress invalid content. Should render failed, edit state.'
+		).toHaveValue( 'https://wordpress.org/gutenberg/handbook/' );
+
+		await embedUtils.insertEmbed( 'https://twitter.com/thatbunty' );
+		await expect(
+			currenEmbedBlock.getByRole( 'textbox', { name: 'Embed URL' } ),
+			'Provider whose oembed API has gone wrong. Should render failed, edit state.'
+		).toHaveValue( 'https://twitter.com/thatbunty' );
+
+		await embedUtils.insertEmbed(
+			'https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/'
+		);
+		await expect(
+			currenEmbedBlock.locator( 'figure' ),
+			'WordPress valid content. Should render valid figure element.'
+		).toHaveClass( 'wp-block-embed' );
+
+		await embedUtils.insertEmbed(
+			'https://www.youtube.com/watch?v=lXMskKTw3Bc'
+		);
+		await expect(
+			currenEmbedBlock.locator( 'figure' ),
+			'Video content. Should render valid figure element, and include the aspect ratio class.'
+		).toHaveClass( /wp-embed-aspect-16-9/ );
```

Comment:
If we strictly migrate the original tests, do we also need to check the `.wp-block-embed` and `.is-type-video` classes?

Reply:
I've altered the assertion based on the error message.

Reply:
I see, that's certainly true 👍